sigle/src/modules/layout/components/HeaderDropdown.tsx

```diff
@@ -0,0 +1,240 @@
+  DropdownMenuSeparator,
+  DropdownMenuTrigger,
+  Button,
+  Box,
+  Typography,
+  StyledChevron,
+  Flex,
+} from '../../../ui';
+import { generateAvatar } from '../../../utils/boringAvatar';
+import {
+  createNewEmptyStory,
+  getStoriesFile,
+  saveStoriesFile,
+  saveStoryFile,
+} from '../../../utils';
+import { useRouter } from 'next/router';
+import { useState } from 'react';
+import { toast } from 'react-toastify';
+import { useQueryClient } from 'react-query';
+import { useGetUserSettings } from '../../../hooks/appData';
+import * as Fathom from 'fathom-client';
+import { Goals } from '../../../utils/fathom';
+import { userSession } from '../../../utils/blockstack';
+import { createSubsetStory } from '../../editor/utils';
+import { styled } from '../../../stitches.config';
+import { useAuth } from '../../auth/AuthContext';
+import { signOut } from 'next-auth/react';
+import {
+  ArchiveIcon,
+  CrumpledPaperIcon,
+  FileTextIcon,
+  MixIcon,
+} from '@radix-ui/react-icons';
+import { useTheme } from 'next-themes';
+import { Switch, SwitchThumb } from '../../../ui/Switch';
+
+const ImageContainer = styled('div', {
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  overflow: 'hidden',
+  width: 24,
+  height: 24,
+  br: '$1',
+});
+
+export const HeaderDropdown = () => {
+  const { data: settings } = useGetUserSettings();
+  const router = useRouter();
+  const { user } = useAuth();
+  const queryClient = useQueryClient();
+  const [loadingCreate, setLoadingCreate] = useState(false);
+  const { resolvedTheme, setTheme } = useTheme();
+
+  const handleCreateNewPrivateStory = async () => {
+    setLoadingCreate(true);
+    try {
+      const storiesFile = await getStoriesFile();
+      const story = createNewEmptyStory();
+
+      storiesFile.stories.unshift(
+        createSubsetStory(story, { plainContent: '' })
+      );
+
+      await saveStoriesFile(storiesFile);
+      await saveStoryFile(story);
+
+      Fathom.trackGoal(Goals.CREATE_NEW_STORY, 0);
+      router.push('/stories/[storyId]', `/stories/${story.id}`);
+    } catch (error) {
+      console.error(error);
+      toast.error(error.message);
+      setLoadingCreate(false);
+    }
+  };
+
+  const handleLogout = () => {
+    queryClient.removeQueries();
+    userSession.signUserOut();
+    signOut();
+  };
+
+  const toggleTheme = () => {
+    resolvedTheme === 'dark' ? setTheme('light') : setTheme('dark');
+  };
+
+  const userAddress =
+    user?.profile.stxAddress.mainnet || user?.profile.stxAddress;
+
+  const upperNavItems = [
+    {
+      name: 'Drafts',
+      path: '/',
+      icon: <CrumpledPaperIcon />,
+    },
+    {
+      name: 'Published',
+      path: '/published',
+      icon: <ArchiveIcon />,
+    },
+    {
+      name: 'Analytics',
+      path: '/analytics',
+      icon: <MixIcon />,
+    },
+  ];
+
+  const lowerNavItems = [
+    {
+      name: 'Settings',
+      path: '/settings',
+    },
+    {
+      name: 'Upgrade',
```

Comment:
Should the upgrade button be hidden if the user does have an active creator + subscription?

![Screenshot 2022-07-22 at 16 18 02](https://user-images.githubusercontent.com/5749437/180458823-a0174843-7f05-41ce-bf04-47e4be3e8220.png)